lib/chromic_pdf/api.ex

```diff
@@ -10,16 +10,36 @@ defmodule ChromicPDF.API do
     Browser,
     CaptureScreenshot,
     GhostscriptPool,
+    OutputOptions,
     PDFOptions,
-    PDFAOptions,
     PrintToPDF
   }
 
   @spec print_to_pdf(
           ChromicPDF.Supervisor.services(),
-          ChromicPDF.source() | ChromicPDF.source_and_options(),
+          ChromicPDF.source()
+          | ChromicPDF.source_and_options()
+          | [ChromicPDF.source() | ChromicPDF.source_and_options()],
           [ChromicPDF.pdf_option() | ChromicPDF.export_option()]
         ) :: ChromicPDF.export_return()
+  def print_to_pdf(services, sources, opts) when is_list(sources) and is_list(opts) do
+    with_tmp_dir(fn tmp_dir ->
+      sources =
+        Enum.map(sources, fn
+          %{opts: source_opts} = source -> %{source | opts: Keyword.merge(source_opts, opts)}
+          source -> source
```

Comment:
The way the options are passed through to `print_tmp` are a bit confusing, IMO. Why not just

```suggestion
          source -> %{source: source, opts: opts}
```

and remove the last argument of `print_tmp/4`?

Reply:
good point here! 👍